src/test/java/com/redislabs/redisgraph/exceptions/JRedisGraphErrorTest.java

```diff
@@ -6,136 +6,190 @@
-        api = new RedisGraph();
-        Assert.assertNotNull(api.query("social", "CREATE (:person{mixed_prop: 'strval'}), (:person{mixed_prop: 50})"));
+  private RedisGraphContextGenerator api;
+
+  @Before
+  public void createApi(){
+    api = new RedisGraph();
+    Assert.assertNotNull(api.query("social", "CREATE (:person{mixed_prop: 'strval'}), (:person{mixed_prop: 50})"));
+  }
+  @After
+  public void deleteGraph() {
+
+    api.deleteGraph("social");
+    api.close();
+  }
+
+  @Test
+  public void testExceptions() {
+    try { 
+      throw new JRedisGraphCompileTimeException("Test message 1");
+    }catch (JRedisGraphCompileTimeException e) {
+      Assert.assertEquals( "Test message 1", e.getMessage());
     }
-    @After
-    public void deleteGraph() {
 
-        api.deleteGraph("social");
-        api.close();
+    Exception cause = new IndexOutOfBoundsException("Index Error");
+    try { 
+      throw new JRedisGraphCompileTimeException("Test message 2", cause);
+    }catch (JRedisGraphCompileTimeException e) {
+      Assert.assertEquals( "Test message 2", e.getMessage());
+      Assert.assertEquals( cause, e.getCause());
     }
 
-    @Rule
-    public ExpectedException exceptionRule = ExpectedException.none();
-
-    @Test
-    public void testSyntaxErrorReporting() {
-        exceptionRule.expect(JRedisGraphCompileTimeException.class);
-        exceptionRule.expectMessage("Type mismatch: expected String but was Integer");
+    try {
+      throw new JRedisGraphCompileTimeException(cause);
+    }catch (JRedisGraphCompileTimeException e) {
+      Assert.assertEquals( "java.lang.IndexOutOfBoundsException: Index Error", e.getMessage());
+      Assert.assertEquals( cause, e.getCause());
+    }
+    
+    try { 
+      throw new JRedisGraphRunTimeException("Test message 3");
+    }catch (JRedisGraphRunTimeException e) {
+      Assert.assertEquals( "Test message 3", e.getMessage());
+    }
 
-        // Issue a query that causes a compile-time error
-        api.query("social", "RETURN toUpper(5)");
+    try { 
+      throw new JRedisGraphRunTimeException("Test message 4", cause);
+    }catch (JRedisGraphRunTimeException e) {
+      Assert.assertEquals( "Test message 4", e.getMessage());
+      Assert.assertEquals( cause, e.getCause());
+    }
 
+    try {
+      throw new JRedisGraphRunTimeException(cause);
+    }catch (JRedisGraphRunTimeException e) {
+      Assert.assertEquals( "java.lang.IndexOutOfBoundsException: Index Error", e.getMessage());
+      Assert.assertEquals( cause, e.getCause());
+    }
+  }
+
+  @Test
+  public void testSyntaxErrorReporting() {
+    try {
+      // Issue a query that causes a compile-time error
+      api.query("social", "RETURN toUpper(5)");
+    } catch (Exception e) {
+      Assert.assertEquals(JRedisGraphCompileTimeException.class, e.getClass());
+      Assert.assertEquals( "redis.clients.jedis.exceptions.JedisDataException: Type mismatch: expected String but was Integer", e.getMessage());
```

Comment:
can you use juint 5 for `assertThrows`?
https://www.baeldung.com/junit-assert-exception

Reply:
we are not using Junit 5